addons/website_mail/static/src/js/website_mail.js

```diff
@@ -0,0 +1,241 @@
+        willStart: function(){
+            var self = this;
+            // load qweb template and init data
+            return $.when(
+                rpc.query({
+                    route: '/website_mail/init',
+                    params: this._messageFetchPrepareParams()
+                }), this._loadTemplates()
+            ).then(function(result){
+                // bind events
+                self.on("change:messages", self, self._renderMessages);
+                self.on("change:message_count", self, function(){
+                    self._renderMessageCount();
+                    self.set('pager', self._pager(self._current_page));
+                });
+                self.on("change:pager", self, self._renderPager);
+                self.on("change:domain", self, self._onChangeDomain);
+                // set options and parameters
+                self.options = _.extend(self.options, result['options'] || {});
+                self.set('message_count', self.options['message_count']);
+                self.set('messages', self.preprocessMessages(result['messages']));
+                return result;
+            });
+        },
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        /**
+         * Fetch the messages nad the message count from the server for the
+         * current page and current domain.
+         *
+         * @params {Array}
+         * @returns {Deferred}
+         */
+        messageFetch: function(domain){
+            var self = this;
+            return rpc.query({
+                route: '/website_mail/fetch',
+                params: self._messageFetchPrepareParams()
+            }).then(function(result){
+                self.set('messages', self.preprocessMessages(result['messages']));
+                self.set('message_count', result['message_count']);
+            });
+        },
+        /**
+         * Update the messages format
+         *
+         * @params {Array<Object>}
+         * @returns {Array}
+         */
+        preprocessMessages: function(messages){
+            _.each(messages, function(m){
+                m['author_avatar_url'] = _.str.sprintf('/web/image/%s/%s/author_avatar/50x50', 'mail.message', m.id);
+                m['published_date_str'] = _.str.sprintf(_t('Published on %s'), moment(m.date).format('MMMM Do YYYY, h:mm:ss a'));
+            });
+            return messages;
+        },
+
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+
+        /**
+         * Change the current page by refreshing current domain
+         *
+         * @private
+         * @param {Number} page
+         * @param {Array} domain
+         */
+        _changeCurrentPage: function(page, domain){
```

Comment:
Yes that for this, but when changing domain, I have to reset the `current_page` to trigger the messages fetch. But if I set the current_page to 1, and I'm already on the page one, replacing `1` by `1` does not trigger the fetching. So I have to use the domain, which is an object to trigger the fetching.